src/Avalonia.Input/DragDropDevice.cs

```diff
@@ -19,11 +19,11 @@ private Interactive GetTarget(IInputElement root, Point local)
             return null;
         }
         
-        private DragDropEffects RaiseDragEvent(Interactive target, RoutedEvent<DragEventArgs> routedEvent, DragDropEffects operation, IDataObject data)
+        private DragDropEffects RaiseDragEvent(Interactive target, IInputElement inputRoot, Point point, RoutedEvent<DragEventArgs> routedEvent, DragDropEffects operation, IDataObject data)
         {
             if (target == null)
                 return DragDropEffects.None;
-            var args = new DragEventArgs(routedEvent, data)
+            var args = new DragEventArgs(routedEvent, data, target, target != null ? inputRoot.TranslatePoint(point, target) : point)
```

Comment:
If `target` is never null, then it should probably be checked as an invariant on entering this method, e.g.

```
Contract.Requires<ArgumentNullException>(target != null);
```

Doing the check here suggests it can sometimes be null.

Reply:
Removed null check. If target is null this code is never called. See the check above:
```
             if (target == null)
                 return DragDropEffects.None;
```